File: app/src/main/java/com/baidu/tv/player/ui/main/PlaylistCardViewHolder.java
package com.baidu.tv.player.ui.main;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.baidu.tv.player.R;
import com.bumptech.glide.Glide;
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions;

/**
 * 播放列表卡片ViewHolder
 */
public class PlaylistCardViewHolder extends RecyclerView.ViewHolder {
    public ImageView ivCover;
    public TextView tvPlaylistName;
    public TextView tvStats;
    public ImageView ivDelete;
    public ImageView ivRefresh;

    public PlaylistCardViewHolder(View itemView) {
        super(itemView);
        ivCover = itemView.findViewById(R.id.iv_cover);
        tvPlaylistName = itemView.findViewById(R.id.tv_playlist_name);
        tvStats = itemView.findViewById(R.id.tv_stats);
        ivDelete = itemView.findViewById(R.id.iv_delete);
        ivRefresh = itemView.findViewById(R.id.iv_refresh);
    }

    /**
     * 绑定数据到视图
     */
    public void bind(String coverImagePath, String playlistName, String stats) {
        // 加载封面图片
        if (coverImagePath != null && !coverImagePath.isEmpty()) {
            Glide.with(ivCover.getContext())
                    .load(coverImagePath)
                    .centerCrop()
                    .transition(DrawableTransitionOptions.withCrossFade())
                    .into(ivCover);
        } else {
            // 使用默认图片
            ivCover.setImageResource(R.drawable.banner);
        }

        // 设置播放列表名称
        tvPlaylistName.setText(playlistName);

        // 设置统计信息
        tvStats.setText(stats);
    }
}

File: app/src/main/java/com/baidu/tv/player/kt/ui/main/PlaylistCardViewHolder.kt
package com.baidu.tv.player.kt.ui.main

import android.net.Uri
import androidx.recyclerview.widget.RecyclerView
import com.baidu.tv.player.kt.R
import com.baidu.tv.player.kt.databinding.ItemPlaylistCardBinding
import com.bumptech.glide.Glide
import com.bumptech.glide.load.model.GlideUrl
import com.bumptech.glide.load.model.LazyHeaders
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions

class PlaylistCardViewHolder(
    val binding: ItemPlaylistCardBinding,
    private val accessTokenProvider: () -> String? = { null },
) : RecyclerView.ViewHolder(binding.root) {

    fun bind(coverImagePath: String?, playlistName: String, stats: String) {
        if (!coverImagePath.isNullOrEmpty()) {
            val cover = if (coverImagePath.startsWith("http", ignoreCase = true)) {
                val uri = Uri.parse(coverImagePath)
                val host = uri.host.orEmpty().lowercase()
                val isBaiduUrl = host == "baidu.com" || host.endsWith(".baidu.com") ||
                    host == "baidupcs.com" || host.endsWith(".baidupcs.com")
                val token = accessTokenProvider()?.takeIf { it.isNotBlank() }
                if (isBaiduUrl && token != null &&
                    uri.getQueryParameter("access_token").isNullOrBlank()
                ) {
                    GlideUrl(
                        uri.buildUpon().appendQueryParameter("access_token", token).build().toString(),
                        LazyHeaders.Builder().addHeader("User-Agent", "pan.baidu.com").build(),
                    )
                } else {
                    coverImagePath
                }
            } else {
                coverImagePath
            }
            Glide.with(binding.ivCover.context)
                .load(cover)
                .placeholder(R.drawable.banner)
                .error(R.drawable.banner)
                .centerCrop()
                .transition(DrawableTransitionOptions.withCrossFade())
                .into(binding.ivCover)
        } else {
            binding.ivCover.setImageResource(R.drawable.banner)
        }
        binding.tvPlaylistName.text = playlistName
        binding.tvStats.text = stats
    }
}
